MekHQ/src/mekhq/campaign/personnel/education/EducationController.java

```diff
@@ -464,88 +444,56 @@ public static void completeEducation(Campaign campaign, Person person) {
+            } catch (Exception e) {
+                travelTime = Math.max(2, campaign.getSimplifiedTravelTime(campaign.getSystemById("Strana Mechty")));
             }
-
-            return null;
+        } else {
+            travelTime = Math.max(2, campaign.getSimplifiedTravelTime(campaign.getSystemById(person.getEduAcademySystem())));
         }
-        return daysOfTravelFrom;
+
+        campaign.addReport(person.getHyperlinkedName() + ' '
+                + String.format(resources.getString("returningFromSchool.text"), travelTime));
+
+        person.setEduDaysOfTravelFromAcademy(travelTime);
+        person.setEduDaysOfTravel(0);
+        person.setEduEducationStage(4);
     }
 
     /**
      * Processes the journey home for a person.
      *
      * @param campaign         the campaign the person is in
      * @param person           the person whose journey home is being processed
-     * @param daysOfEducation  the number education days remaining
-     * @param daysOfTravelFrom the number of days it takes for the person to travel from the campaign location to the unit
      */
-    private static void processJourneyHome(Campaign campaign, Person person, Integer daysOfEducation, Integer daysOfTravelFrom) {
-        if ((daysOfEducation == 0) && (daysOfTravelFrom > 0)) {
-            int travelTime = 0;
-            boolean clanException = false;
+    private static void processJourneyHome(Campaign campaign, Person person) {
+        int travelTime = 0;
 
-            try {
-                travelTime = campaign.getSimplifiedTravelTime(campaign.getSystemById(person.getEduAcademySystem()));
-            } catch (Exception e) {
-                clanException = true;
-            }
+        try {
+            travelTime = Math.max(2, campaign.getSimplifiedTravelTime(campaign.getSystemById(person.getEduAcademySystem())));
 
-            if (clanException) {
-                person.setEduDaysOfTravelFromAcademy(daysOfTravelFrom - 1);
-                person.setEduDaysOfTravel(person.getEduDaysOfTravel() + 1);
-            } else {
-                if (travelTime > daysOfTravelFrom) {
-                    person.setEduDaysOfTravelFromAcademy(travelTime);
-                } else if (travelTime < daysOfTravelFrom) {
-                    person.setEduDaysOfTravelFromAcademy(travelTime / 2);
-                    person.setEduDaysOfTravel(0);
-                } else {
-                    person.setEduDaysOfTravelFromAcademy(daysOfTravelFrom - 1);
-                    person.setEduDaysOfTravel(person.getEduDaysOfTravel() + 1);
-                }
+            if (travelTime != person.getEduDaysOfTravel()) {
+                person.setEduDaysOfTravelFromAcademy(travelTime);
             }
 
-            if ((person.getEduDaysOfTravelFromAcademy()) == 0) {
-                person.changeStatus(campaign, campaign.getLocalDate(), PersonnelStatus.ACTIVE);
-            }
+            person.setEduDaysOfTravel(person.getEduDaysOfTravel() + 1);
```

Comment:
This could be a `finally` block, since the same operation happens either way, and at the end.